tests/FSharp.Test.Utilities/Compiler.fs

```diff
@@ -824,6 +827,30 @@ module rec Compiler =
         | CS cs -> compileCSharp cs
         | _ -> failwith "TODO"
 
+    let compileExisting (cUnit: CompilationUnit) : CompilationResult =
```

Comment:
This relies on file sitting on the disk. Everything else here either generates stuff on the fly or does some copying, creating new files anyway.

Since the `CachingDriver` relies on the timestamps, to test the recompilation we need to simulate the files "just sitting there", that's what it is for.

Reply:
Yes. But unfortunately, there is still some data copying/generation going on. Usually in [compileDisposable](https://github.com/dotnet/fsharp/blob/main/tests/FSharp.Test.Utilities/CompilerAssert.fs#L443). A few remaining occasions work via `compile` but they still [generate](https://github.com/dotnet/fsharp/blob/main/tests/FSharp.Test.Utilities/CompilerAssert.fs#L510) output path, which we need to have stable in these tests.

We really just need to call `rawCompile` with given source path and given output path and surprisingly there are no test means for that yet.